### PR TITLE
chore(perf): use pyenv to install

### PR DESCRIPTION
**Pull request by majorgreys**
Use pyenv to install Python. This is necessary for our performance testing as the binaries in the official docker python images have had their symbols stripped (https://github.com/docker-library/python/pull/483).

### CODE REVIEW

File: benchmarks/base/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,27 @@
...
+    make build-essential libssl-dev zlib1g-dev \
+    libbz2-dev libreadline-dev libsqlite3-dev wget curl llvm \
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev \
+    git ca-certificates \
+    # cleaning up unused files
+    && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+    && rm -rf /var/lib/apt/lists/*
+ENV PYENV_ROOT "/pyenv"
+ENV PATH "$PYENV_ROOT/bin:$PATH"
+RUN git clone https://github.com/pyenv/pyenv.git "$PYENV_ROOT"
```
Comment (brettlangdon) [inline benchmarks/base/Dockerfile:14]: should we use a specific git tag or something?

File: benchmarks/base/Dockerfile [hunk 2]
```diff
@@ -1,6 +1,27 @@
...
+RUN pyenv install "$PYTHON_VERSION"
+
+FROM debian:buster-slim
```
Comment (brettlangdon) [inline benchmarks/base/Dockerfile:17]: any reason to have as multi-build? doesn't seem like there is much to strip away, a few `lib*-dev` dependencies, but shouldn't make a big deal with the image?

Comment (majorgreys) [reply benchmarks/base/Dockerfile:19]: The difference is bigger than I would have expected: 821mb without multistage, 580mb with multistage.

Comment (brettlangdon) [reply benchmarks/base/Dockerfile:19]: /shrug alright, interesting!

File: benchmarks/base/Dockerfile [hunk 3]
```diff
@@ -1,6 +1,29 @@
...
+    libncursesw5-dev xz-utils tk-dev libxml2-dev libxmlsec1-dev libffi-dev liblzma-dev \
+    git ca-certificates \
+    # cleaning up unused files
+    && apt-get purge -y --auto-remove -o APT::AutoRemove::RecommendsImportant=false \
+    && rm -rf /var/lib/apt/lists/*
+ENV PYENV_ROOT "/pyenv"
+WORKDIR "$PYENV_ROOT"
+ENV PATH "$PYENV_ROOT/bin:$PATH"
+RUN git clone -n https://github.com/pyenv/pyenv.git "$PYENV_ROOT"
+RUN git checkout cef86ce462674c726cb60ddd0e19da0bf39a27c9 # v2.0.4
```
Comment (brettlangdon) [inline benchmarks/base/Dockerfile:16]:
```suggestion
RUN git clone --depth 1 https://github.com/pyenv/pyenv.git --branch "v$PYENV_VERSION" --single-branch
```

File: benchmarks/base/Dockerfile [hunk 4]
```diff
@@ -1,6 +1,29 @@
-ARG PYTHON_VERSION=3.9-slim-buster
+FROM debian:buster-slim as base
 
-FROM python:${PYTHON_VERSION} as python
+ARG PYTHON_VERSION=3.9.6
```
Comment (brettlangdon) [inline benchmarks/base/Dockerfile:3]:
```suggestion
ARG PYTHON_VERSION=3.9.6
ARG PYENV_VERSION=2.0.4
```

**Review by brettlangdon — APPROVED** [review]

**Review by nizox — APPROVED** [review]

**Pull request merged**